Initial global build PR pipeline

trylek opened the pull request:
As of right now this change just runs the root "build" script on
three platforms. I'm open to suggestions as to the desired level
of actual testing and / or to the platform set.

trylek committed:
Initial global build PR pipeline

Fixes: #951

trylek committed:
Fix OSX build dependencies and container use per Santiago's advice

jaredpar commented on eng/pipelines/global-build.yml:
```diff
@@ -0,0 +1,51 @@
+trigger: none
```
Can we add a comment at the top here to state the purpose of this job? Essentially that it serves to validate that changes don't break the local developer build workflow.

trylek replied:
Good point, fixed.

trylek committed:
Comment the purpose of the new pipeline per Jared's PR feedback

dagood commented on eng/pipelines/common/global-build-job.yml:
```diff
@@ -0,0 +1,47 @@
+    steps:
+    - ${{ if eq(parameters.osGroup, 'OSX') }}:
+      - script: |
+          brew install pkgconfig icu4c openssl
```
I'd add `set -ex` for stop on error and to show the commands in the logs as they're run.

trylek replied:
Fixed.

dagood commented on eng/pipelines/common/global-build-job.yml:
```diff
@@ -0,0 +1,47 @@
+  container: ''
+  testGroup: ''
+  crossrootfsDir: ''
+  timeoutInMinutes: ''
```
Trim unused variables (to avoid a false assumption that they work when someone wants to add more legs later)?

trylek replied:
Fixed.

dagood commented on eng/pipelines/common/global-build-job.yml:
```diff
@@ -0,0 +1,47 @@
+      value: "-c Release"
+    - ${{ if eq(parameters.buildConfig, 'debug') }}:
+      - name: configArg
+        value: "-c Debug"
```
Could you just do this, or does capitalization matter?

```yaml
    - name: configArg
      value: '-c ${{ parameters.buildConfig }}'
```

safern replied:
I think we should instead pass down as uppercase values into the template. When calling build.cmd/.sh in the root, we expect uppercase values because we just call MSBuild, not like src/coreclr/build.cmd/sh which actually expects lowercase arguments as `-release, -checked, -debug`.

safern commented on eng/pipelines/global-build.yml:
```diff
@@ -0,0 +1,55 @@
+    include:
+    - '*'
+    - docs/manpages/*
+    exclude:
```
What about excluding `eng\pipelines\coreclr`, `eng\pipelines\libraries`, `eng\pipelines\` and including `eng\pipelines\global-build.yml`?

trylek replied:
Fixed.

trylek committed:
Prune unused parameters of global-build-job

trylek committed:
Fix variable propagation

trylek commented:
OK, I believe I have addressed all review feedback, the new pipeline shows as green and the above failures are unrelated to my change. Is there a chance to get this reviewed so that I can merge it in before the holiday break? I believe that, even in its limited form the job is relatively useful for regression testing of Jeremy's local live-live build and it's relatively fast (under one hour).

Thanks and happy holidays everyone!

Tomas

trylek commented:
Thanks Davis!
